tests/components/renault/test_init.py

```diff
@@ -1,85 +1,63 @@
 """Tests for Renault setup process."""
-from unittest.mock import AsyncMock, patch
+from unittest.mock import patch
 
 import aiohttp
-import pytest
 from renault_api.gigya.exceptions import InvalidCredentialsException
-from renault_api.kamereon import schemas
 
-from homeassistant.components.renault import (
-    RenaultHub,
-    async_setup_entry,
-    async_unload_entry,
-)
 from homeassistant.components.renault.const import DOMAIN
-from homeassistant.components.renault.renault_vehicle import RenaultVehicleProxy
-from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.config_entries import ConfigEntryState
 
-from .const import MOCK_CONFIG
+from . import get_mock_config_entry, setup_renault_integration_simple
 
-from tests.common import MockConfigEntry, load_fixture
 
-
-async def test_setup_unload_and_reload_entry(hass):
+async def test_setup_unload_entry(hass):
     """Test entry setup and unload."""
-    # Create a mock entry so we don't have to go through config flow
-    config_entry = MockConfigEntry(
-        domain=DOMAIN, data=MOCK_CONFIG, entry_id="test", unique_id=123456
-    )
-    renault_account = AsyncMock()
-    renault_account.get_vehicles.return_value = (
-        schemas.KamereonVehiclesResponseSchema.loads(
-            load_fixture("renault/vehicle_zoe_40.json")
-        )
-    )
-
-    with patch("renault_api.renault_session.RenaultSession.login"), patch(
-        "renault_api.renault_client.RenaultClient.get_api_account",
-        return_value=renault_account,
-    ):
-        # Set up the entry and assert that the values set during setup are where we expect
-        # them to be.
-        assert await async_setup_entry(hass, config_entry)
-        assert DOMAIN in hass.data and config_entry.unique_id in hass.data[DOMAIN]
-        assert isinstance(hass.data[DOMAIN][config_entry.unique_id], RenaultHub)
+    with patch("homeassistant.components.renault.PLATFORMS", []):
+        config_entry = await setup_renault_integration_simple(hass)
 
-        renault_hub: RenaultHub = hass.data[DOMAIN][config_entry.unique_id]
-        assert len(renault_hub.vehicles) == 1
-        assert isinstance(
-            renault_hub.vehicles["VF1AAAAA555777999"], RenaultVehicleProxy
-        )
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+    assert config_entry.state is ConfigEntryState.LOADED
+    assert config_entry.unique_id in hass.data[DOMAIN]
```

Comment:
Side note: We normally use the config entry entry_id as key in hass.data, not the config entry unique_id.

Reply:
One for a follow-up PR?

Reply:
Yes, please.

Reply:
I'll add it to #53680 as soon as this one is merged in.